src/BusinessLayer/InternetReporting.cpp

```diff
@@ -17,278 +17,355 @@
-                                     const MpptData& mpptData,
-                                     const LightsData& lightsData,
-                                     const AuxBmsData& auxBmsData,
+                                     DataContainer& dataContainer0,
+                                     DataContainer& dataContainer1,
                                      InternetView& view)
 
     : communicationService_(commService)
-    , keyMotorData_(keyMotorData)
-    , motor0DetailsData_(motor0DetailsData)
-    , motor1DetailsData_(motor1DetailsData)
-    , driverControlsData_(driverControlsData)
-    , motorFaultsData_(motorFaultsData)
-    , batteryFaultsData_(batteryFaultsData)
-    , batteryData_(batteryData)
-    , mpptData_(mpptData)
-    , lightsData_(lightsData)
-    , auxBmsData_(auxBmsData)
+    , keyMotorData0_(dataContainer0.getKeyMotorData())
+    , motor0DetailsData0_(dataContainer0.getMotor0DetailsData())
+    , motor1DetailsData0_(dataContainer0.getMotor1DetailsData())
+    , driverControlsData0_(dataContainer0.getDriverControlsData())
+    , motorFaultsData0_(dataContainer0.getMotorFaultsData())
+    , batteryFaultsData0_(dataContainer0.getBatteryFaultsData())
+    , batteryData0_(dataContainer0.getBatteryData())
+    , mpptData0_(dataContainer0.getMpptData())
+    , lightsData0_(dataContainer0.getLightsData())
+    , auxBmsData0_(dataContainer0.getAuxBmsData())
+    , keyMotorData1_(dataContainer1.getKeyMotorData())
+    , motor0DetailsData1_(dataContainer1.getMotor0DetailsData())
+    , motor1DetailsData1_(dataContainer1.getMotor1DetailsData())
+    , driverControlsData1_(dataContainer1.getDriverControlsData())
+    , motorFaultsData1_(dataContainer1.getMotorFaultsData())
+    , batteryFaultsData1_(dataContainer1.getBatteryFaultsData())
+    , batteryData1_(dataContainer1.getBatteryData())
+    , mpptData1_(dataContainer1.getMpptData())
+    , lightsData1_(dataContainer1.getLightsData())
+    , auxBmsData1_(dataContainer1.getAuxBmsData())
     , view_(view)
+    , packetNum(0)
 {
-//    connect(&view_, SIGNAL(sendAll()), this, SLOT(sendAll()));
+    connect(&view_, SIGNAL(sendAll(int)), this, SLOT(sendAll(int)));
 }
 
 QJsonArray InternetReporting::makeKeyMotor()
 {
+    const KeyMotorData* keyMotorData_;
+    if (packetNum == 0)
+    {
+        keyMotorData_ = &keyMotorData0_;
+    }
+    else
+    {
+        keyMotorData_ = &keyMotorData1_;
+    }
+
     QJsonArray keyMotor;
     QJsonObject keyMotor0;
-    keyMotor0.insert("Alive", keyMotorData_.motor0Alive());
-    keyMotor0.insert("SetCurrent", keyMotorData_.motor0SetCurrent());
-    keyMotor0.insert("SetVelocity", keyMotorData_.motor0SetVelocity());
-    keyMotor0.insert("BusCurrent", keyMotorData_.motor0BusCurrent());
-    keyMotor0.insert("BusVoltage", keyMotorData_.motor0BusVoltage());
-    keyMotor0.insert("VehicleVelocity", keyMotorData_.motor0VehicleVelocity());
+    keyMotor0.insert("Alive", keyMotorData_->motor0Alive());
```

Comment:
Going back to the dataContainer point, it can be rewritten as something like
`keyMotor.0.insert("Alive", dataContainerList[packetNum].getKeyMotorData()->motor0Alive())`